### ISSUE TITLE
Contacts Missing

### ISSUE THREAD

**Issue by sybercorp** [signalapp/Signal-Android#8639]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
There are many Issues here for this exact problem, spanning the last few years, and it is still not fixed.  Some contacts do not show in Signal (like they do in other messaging apps, or in the Contacts app).  The missing contacts are seemingly random and are not Signal contacts - just standard contacts with a phone number to message via SMS/MMS.  If you search for these contacts in Signal, they are not found.  If you start a message and just type in their number manually, then send the message, they show up in the conversations list as their name (as in, the app matches their number to a name, proving that it does have Contacts permissions.  Toggling the Contacts permission off/on does nothing to fix this.  Refreshing the contacts list in Signal does nothing to fix this.  Uninstalling/reinstalling Signal does nothing to fix this.

### Steps to reproduce
- Locate contact that is missing in the contact list
- Start a message to this person by phone number directly
- See the person listed by name in the conversations list
- Notice the person is STILL not listed in the contact list, even though you've just proven Signal can match name to number successfully

**Actual result:** Contact is missing
**Expected result:** Contact is present

### Device info

**Device:** Samsung Note 9
**Android version:** 8.1.0
**Signal version:** 4.34.8

### Link to debug log

https://debuglogs.org/0c4daed3696621100d363c997a042c79a04009bc47cc6a643d1bc4bbf1fca04a

**Comment by iamhshaikh** [comment]
I am facing to I installed signal on my phone and then I installed it on my brother's phone and verified his phone number, but when I go to start conversation, I don't see my contact on his phone I have given all permission and also I don't see his name in my contact list even he is on signal, both are having same problem, I tried to message using phone number directly from his phone and I got his name on that chat but in his phone it shows only my number instead of name as he saved in his contact list, fix this issue ASAP

**Comment by sybercorp** [comment]
This still happen after upgrading to Android Pie, for the record.

**Comment by CueHD** [comment]
In your phone contacts app, find the display settings. You should have the option to show all contacts as well as contacts from different accounts (phone, Google, Signal, etc.). Try the different selections and see if there is one (other than Signal) that seems to make sense with what you are seeing from Signal. I wonder if the problem is related to the syncing of your contacts between the various accounts.

**Comment by sybercorp** [comment]
All of my contacts are stored on the same Google account.  I've already verified that.  I've also deleted the contacts I've had this happen with and re-added them (which didn't make a difference).

I've forced contact syncing numerous times and it makes no difference.  If I create test contacts and then sync contacts, the test contacts show up but the problematic ones still don't.

Have you reviewed the debug output I provided? Did it contain anything useful?

Jesse Peden
Wire ID: jessepeden
sybercorp@gmail.com
(Sent from a mobile device - please excuse any typos or brevity)

________________________________

**Comment by sybercorp** [comment]
Update: I've managed to get one of the missing contacts to appear in Signal by un-merging that contact's entries.  Even though the primary entry had the same phone number directly on it as the merged entries, until I un-merged then to leave only the primary entry, the contact would not show up in Signal.  Even after un-merging, there was a delay.  I had told Signal to refresh Contacts manually, which it appeared to do, but the contact was still missing.  A few minutes later I noticed it was finally showing.  I will test more of my contacts that have merged entries to see if that is truly the problem (that Signal doesn't like merged contact entries, where other programs have no problem reading them).

**Comment by sybercorp** [comment]
Update 2: I have other contacts with exactly the same configuration as the problematic contacts before un-merging them, and they show up in Signal without any modifications.  So, it appears to maybe be random.

**Comment by BigSweetPotato** [comment]
I ran into this issue as well. The missing contacts seem to be random. They are all saved under Google account.

Version: 4.39.4

**Comment by carlin-q-scott** [comment]
Seems like there's a known fix for the issue but it was never fully implemented: https://github.com/signalapp/Signal-Android/issues/7394#issuecomment-363164293. A bot closed and locked that issue for some reason.

**Comment by nekromoff** [comment]
Still an issue for me as well.

**Comment by sybercorp** [comment]
@nekromoff That may have been a workaround to that particular person's issue but that is not a true solution.  That was posted 3+ years ago and it still occurs to users, which is pretty sad, considering that the ability to see contacts in order to send them messages us literally THE primary function of this (or any) messaging app.

**Comment by ProactiveServices** [comment]
Following the instructions in issue 4338 hasn't resolved this for me.

**Comment by bljusruk** [comment]
Could not make signal have the same list of contacts as my phone and my Google Account. Tried different things advised here and in other places on the net - nothing worked. At the end stopped using signal.

**Comment by danielsokolowski** [comment]
Impacted by this issue as well, just as described in the issue, manually you can SMS message the contact and it shows up in search in the SingalApp but the contact itself does not.

**Comment by jpodpro** [comment]
having this problem too. why do chat apps try to act like phones? can't we just have a simple chat app that lets you add contacts manually without having to pretend like they are a phone? ffs this is maddening.

**Comment by kjongsma** [comment]
Same problem here. Have tried deleting account and resyncing to no avail. Very annoying! Fortunately, sending does work when initiating from Contacts app.

**Comment by greatleapingcrab** [comment]
Same issue here. Thanks to kjongsma for the pointer re using a Contacts app, but this is a bit of a  crappifying bug.

**Comment by kriswdue** [comment]
I too have this issue, and I've had it ever since i started using Signal a year ago. I've beeen getting by using kjongsma's workaround. 

I am now contemplating using signal only for Signal contacts, and reverting to standard sms app.

Phone: Xiaomi Mi A1, Android Pie.

**Comment by k1o5** [comment]
Same issue. New contacts I add seem to be available to signal, just not most of the existing contacts.

**Comment by johnlb** [comment]
So I just had a problem that acted in a similar way. It turned out that Signal had gotten removed as the primary SMS app and because of that, it didn't show non-Signal contacts when I went to start a new conversation. I could, however, use the search bar in the main window to find them, which is how I figured it out: it gives a message saying you need to switch your SMS app to Signal and gives a button to press. Hope that helps at least some of you!

**Comment by LeSyl21** [comment]
Samedi issue for 3 years... today i will leave Signal... i'm sad everything elle was great

**Comment by sebastientaggart** [comment]
Just chiming in here as well, some contacts don't appear, seemingly at random.

**Comment by aiguofer** [comment]
Same issue here... Getting kind of annoying. Interestingly, the ones that don't show up when searching for a single contact do seem to show up when looking for contacts in a group

**Comment by wrldtvlr** [comment]
More testing to do, but it appears that the latest android release has fixed the missing contact problem. The fix is a little weird as Signal is now showing all contact phone numbers, not just mobile numbers, but I can live with that. It's also possible that the previously missing contacts were ones with multiple phone numbers.

**Comment by greyson-signal** [comment]
@wrldtvlr Oh that's good, I was hoping the latest release would help with this. We were using some relatively complex system-contact queries for showing non-signal contacts in that list. 4.48.x switched to using our locally-synced copy of your system contacts. So now if contacts are missing, it should at least be consistent across the app, and bugs should be more centralized and/or easier to fix.

**Comment by nekromoff** [comment]
I can confirm this as well. I see the missing contacts from before.

**Comment by greyson-signal** [comment]
Perfect! Let's call this particular contacts issue (contacts being recognized but not appearing in search/conversation list) fixed, since that's what the original issue was about.

**Comment by carlin-q-scott** [comment]
Do I need to re-link my phone number to get the fix? I updated to the latest release and the problem is still there.

**Comment by wrldtvlr** [comment]
I didn't need to, but you might try refreshing your contacts when in the compose screen.

**Comment by carlin-q-scott** [comment]
Apparently, I had to set Signal as my SMS/MMS app again after having given up on using it for that. The issue seems fixed!

**Comment by GeeWee** [comment]
What fixed the issue for me was writing a new message and then pulling down to refresh the contacts list.

**Comment by LeeBinder** [comment]
same issue here. Any working solution? None of the tips from below solved this issue:
~~
Open Signal, tap the blue pencil ✏️ button, pull down the contact list to refresh it and see if the contacts are correctly associated again
~~
Android Settings > Accounts > Signal > Menu > Remove Account > then Resync

Note that removing the account will result in the loss of all message threads and group conversations.
~~
Re-registere by switching off and back on:

Signal Settings -> Advanced Settings > Free private messaging and calling
~~

**Comment by LeeBinder** [comment]
possible solution see [my post here](https://github.com/signalapp/Signal-Android/issues/9861#issuecomment-664689160) (worked for me).

**Comment by SexyPantz** [comment]
> Seeing as how this issue has continued for literal years with no one seemingly giving any f..ks. I'm uninstalling Signal and I'm gonna find something else to use.
> 
> I think Signal's plan is to lose all users by attrition.
> 
> Some advice though: instead of just having basic ongoing bugs causing users to finally give up, you guys should just release an update that completely breaks the app. It'll be a more efficient way of losing users rather than this policy of just not caring enough to fix basic problems.
> 
> Thank you for all the hard work you guys put into not working out seemingly simple issues. It takes true dedication to stare into the face of a problem and state defiantly: "I Do Not Care!" (or to more likely to say: "...meh" while ignoring any criticism or suggestion).

I wholeheartedly agree. Signal runs like bare bones bug ridden alpha version, which wound not be a huge deal if it was marketed as a early access or open beta or something by the dev, but that is  not the case. The contacts are buggy (which on a f**king messaging app is kind of huge deal), the UX design is a bit odd and it lacks basic features like changing devices, changing phone numbers and markdown  text formatting.. things that every other platform out there has. Just to be clear, my contacts are not syncing and rather than putting up a fix or a quick troubleshoot guide on the the signa homepage im scuring the depgsst of github like this is some sort of

**Comment by carlosg66** [comment]
I had this issue closed the desktop app and reopened it, ensure phone is unlocked and wait and wait and wait. The contacts will eventually appear. Hope this helps.

**Comment by carlosg66** [comment]
Forgot to add you may need to click the import contact button again on the desktop app

**Comment by RushikeshJoshi4** [comment]
> I had this issue closed the desktop app and reopened it click import contacts again, ensure phone is unlocked and wait and wait and wait. The contacts will eventually appear. Hope this helps.

Yes. Closing signal desktop app on my computer and retrying worked. Thanks.

**Comment by Tanjim131** [comment]
This issue still persists on iOS 14.3

**Comment by MajestyZ** [comment]
This exact same issue still persists on my OnePlus 8 Pro (fully updated). There's one contact that won't show up on my phone, and when she chats with me, I get the notification I can only send her texts. At the same time when I use the desktop application (Windows), I can chat with her via Signal + she shows up in the contact list. 

I've tried deleting the contact, re-adding, refreshing contactslist, clearing data/cache, checked for any inconsistencies in contact account (they're all stored in my Google account). I've got no clue how to fix this. Anyone else running into the same problems still?
